java/team25core/FourWheelDirectIMUDrivetrain.java

```diff
@@ -0,0 +1,388 @@
+        frontRight.setDirection(DcMotor.Direction.REVERSE);
+        rearRight.setDirection(DcMotor.Direction.REVERSE);
+    }
+
+    public void setNoncanonicalMotorDirection()
+    {
+        // This reverses the direction of the drivetrain.
+        frontLeft.setDirection(DcMotorSimple.Direction.REVERSE);
+        rearLeft.setDirection(DcMotorSimple.Direction.REVERSE);
+        frontRight.setDirection(DcMotor.Direction.FORWARD);
+        rearRight.setDirection(DcMotor.Direction.FORWARD);
+    }
+
+    public void setSplitPersonalityMotorDirection(boolean isTeleop)
+    {
+        if (isTeleop == true) {
+            frontLeft.setDirection(DcMotorSimple.Direction.REVERSE);
+            frontRight.setDirection(DcMotorSimple.Direction.FORWARD);
+            rearLeft.setDirection(DcMotorSimple.Direction.FORWARD);
+            rearRight.setDirection(DcMotorSimple.Direction.REVERSE);
+        } else {
+            frontLeft.setDirection(DcMotorSimple.Direction.FORWARD);
+            frontRight.setDirection(DcMotorSimple.Direction.REVERSE);
+            rearLeft.setDirection(DcMotorSimple.Direction.REVERSE);
+            rearRight.setDirection(DcMotorSimple.Direction.FORWARD);
+        }
+
+    }
+
+    @Override
+    public void resetEncoders()
+    {
+        rearLeft.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        rearRight.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        frontLeft.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        frontRight.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+    }
+
+    @Override
+    public void encodersOn()
+    {
+        rearLeft.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rearRight.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        frontLeft.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        frontRight.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+    }
+
+    @Override
+    public void logEncoderCounts()
+    {
+        RobotLog.i("POS Counts FL %d, FR %d, RL %d, RR %d", frontLeft.getCurrentPosition(), frontRight.getCurrentPosition(), rearLeft.getCurrentPosition(), rearRight.getCurrentPosition());
+    }
+
+    @Override
+    public void straight(double speed)
+    {
+        frontRight.setPower(speed);
+        frontLeft.setPower(speed);
+        rearRight.setPower(speed);
+        rearLeft.setPower(speed);
+    }
+    @Override
+    public void strafe(double speed)
+    {
+        double yawError;
+        double yawCorrection;
+
+        yawError = targetYaw - currentYaw;
+        // FIXME We need to constrain this value so it does not flip over
+        // if it gets too big or too small.
+        yawCorrection = yawError * PROPORTIONAL_CONSTANT
+                + currentYawRate * DERIVATIVE_CONSTANT;
+
+        frontRight.setPower(-speed - yawCorrection);
+        rearRight.setPower(speed - yawCorrection);
+        frontLeft.setPower(speed + yawCorrection);
+        rearLeft.setPower(-speed + yawCorrection);
```

Comment:
Ok, I think I see what you are doing here in the context of my previous comments with the tasks.

Couldn't all of this have been accomplished with a drivetrain that implements IMU correction and no changes to the DeadReckonTask, and/or no new copy/paste of the DeadReckonTask?   e.g. Move all IMU initialization into the drivetrain class?